### PR TITLE
Pvc resize UI

### PR DESCRIPTION
**Pull request by am-agrawa**
Fixes: #2524

### CODE REVIEW

File: ocs_ci/ocs/ui/base_ui.py [hunk 1]
```diff
@@ -195,6 +201,45 @@ def take_screenshot(self):
         self.driver.save_screenshot(filename)
         time.sleep(0.5)
 
+    def do_clear(self, locator, timeout=30):
+        """
+        Clear the existing text from UI
+
+        locator (set): (GUI element needs to operate on (str), type (By))
+        timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
```
Comment (prsurve) [inline ocs_ci/ocs/ui/base_ui.py:210]: doc string missing

File: ocs_ci/ocs/ui/base_ui.py [hunk 2]
```diff
@@ -195,6 +201,45 @@ def take_screenshot(self):
...
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((locator[1], locator[0])))
+        element.clear()
+
+    def wait_for_element(self, locator, text_):
+        """
+        Method to wait for a web element to be found
+
+        locator (set): (GUI element needs to operate on (str), type (By))
```
Comment (prsurve) [inline ocs_ci/ocs/ui/base_ui.py:220]: this param should be in args

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 3]
```diff
@@ -96,6 +97,81 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
...
+    def pvc_resize_ui(self, pvc_name, pvc_size, new_size, sc_type):
+        """
+        Resizing pvc via UI
+
+
+        pvc_name (str): the name of pvc
+        pvc_size (str): the size of pvc (GB)
+        new_size (int): the new size of pvc (GB)
+        sc_type (str): storage class type
+
```
Comment (prsurve) [inline ocs_ci/ocs/ui/pvc_ui.py:109]: doc string misssing

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 4]
```diff
@@ -96,6 +97,81 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
...
+        logger.info("Select openshift-storage project")
+        self.do_click(self.pvc_loc["pvc_project_selector"])
+        self.do_click(self.pvc_loc["select_openshift-storage_project"])
+
+        self.do_send_keys(self.pvc_loc["search_pvc"], text=pvc_name)
+
+        logger.info(f"Go to PVC {pvc_name} Page")
+        self.do_click(self.pvc_loc[pvc_name])
+
+        if sc_type in "ocs-storagecluster-ceph-rbd":
```
Comment (prsurve) [inline ocs_ci/ocs/ui/pvc_ui.py:122]: make use of constant here

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 5]
```diff
@@ -96,6 +97,81 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
...
+        pvc_size (str): the size of pvc (GB)
+        new_size (int): the new size of pvc (GB)
+        sc_type (str): storage class type
+
+        """
+        self.navigate_persistentvolumeclaims_page()
+
+        logger.info("Select openshift-storage project")
+        self.do_click(self.pvc_loc["pvc_project_selector"])
+        self.do_click(self.pvc_loc["select_openshift-storage_project"])
```
Comment (jilju) [inline ocs_ci/ocs/ui/pvc_ui.py:115]: Use a test project for creating PVC.

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 4 above)
Comment (jilju) [reply ocs_ci/ocs/ui/pvc_ui.py:122]: Pod is needed for the completion of RBD PVC expansion(both thin and thick PVC). But no need to skip pod creation for cephfs PVC.

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 6]
```diff
@@ -96,6 +97,81 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
...
+        logger.info("Click on Actions")
+        self.do_click(self.pvc_loc["pvc_actions"])
+
+        logger.info("Click on Expand PVC")
+        self.do_click(self.pvc_loc["expand_pvc"])
+
+        logger.info("Clearing the size of existing pvc")
+        self.do_clear(self.pvc_loc["resize-value"])
+
+        if new_size > int(pvc_size):
```
Comment (jilju) [inline ocs_ci/ocs/ui/pvc_ui.py:137]: Ideally, the new size passed from the test case should be greater than the current size. If the check is still necessary, use assert at the beginning of this function.
```
assert new_size > int(pvc_size), (f"New size of the PVC cannot be less than existing size: {new_size}")
```

Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: done

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 5 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: done

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 4 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: done

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 3 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: added

File: ocs_ci/ocs/ui/base_ui.py (hunk 2 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py]: added

File: ocs_ci/ocs/ui/base_ui.py (hunk 1 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py]: added

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 7]
```diff
@@ -87,28 +98,79 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
...
 
         """
         self.navigate_persistentvolumeclaims_page()
 
-        logger.info("Select openshift-storage project")
+        logger.info(f"Select test project {project_name}")
         self.do_click(self.pvc_loc["pvc_project_selector"])
-        self.do_click(self.pvc_loc["select_openshift-storage_project"])
+        self.do_send_keys(self.pvc_loc["search-project"], text=project_name)
+        self.do_click((f"//a[normalize-space()='{project_name}']", By.XPATH))
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/pvc_ui.py:171]: please move this element to views.py

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 8]
```diff
@@ -19,22 +20,29 @@ def __init__(self, driver):
...
         self.navigate_persistentvolumeclaims_page()
 
-        logger.info("Select openshift-storage project")
+        logger.info(f"Search test project {project_name}")
         self.do_click(self.pvc_loc["pvc_project_selector"])
-        self.do_click(self.pvc_loc["select_openshift-storage_project"])
+        self.do_send_keys(self.pvc_loc["search-project"], text=project_name)
+
+        logger.info(f"Select test project {project_name}")
+        self.do_click((f"//a[normalize-space()='{project_name}']", By.XPATH))
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:45]: Not sure if we can access project_name in views.py file, because it is dynamic and name of the project changes every time, I will then have to create an object of TestPvcUserInterface class of test_pvc_ui file inside views.py file. Please suggest a suitable way to do so.

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 7 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:171]: Not sure if we can access project_name in views.py file, because it is dynamic and name of the project changes every time, I will then have to create an object of TestPvcUserInterface class of test_pvc_ui file inside views.py file. Please suggest a suitable way to do so.

**Review by am-agrawa — COMMENTED** [review]
Changed pod_dict_path to pod_dict, refer bug issues-2524

File: ocs_ci/ocs/ui/base_ui.py [hunk 9]
```diff
@@ -195,6 +202,86 @@ def take_screenshot(self):
...
+        try:
+            wait.until(
+                ec.text_to_be_present_in_element(
+                    (locator[1], locator[0]), text_=expected_text
+                )
+            )
+            logger.info(f"Element text found: {expected_text} ")
+            return f"{expected_text}"
+        except TimeoutException:
+            logger.error(f"Could not find expected text: {expected_text} ")
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/base_ui.py:283]: what should be returned if the exception is caught?

Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:283]: Logger error will be raised along with TimeoutException. In my case, since I am using it in Assertion, so Assertion will also fail and execution will stop with AssertionError.

Comment (jilju) [reply ocs_ci/ocs/ui/base_ui.py:283]: The return value will be *None* if TimeoutException is occurred.

Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:248]: Thanks @ebenahar for pointing it out, please review the modifications.

Comment (ebenahar) [reply ocs_ci/ocs/ui/base_ui.py:248]: thanks, I see it's changed and looks good to me

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 7 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: @OdedViner Done

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: @OdedViner Done

File: tests/ui/test_pvc_ui.py [hunk 10]
```diff
@@ -130,11 +145,63 @@ def test_create_delete_pvc(
...
+        pvc_ui_obj.pvc_resize_ui(
+            pvc_name=pvc_name, new_size=new_size, project_name=project_name
+        )
+
+        logger.info("Verifying New Capacity after Pvc Resize")
+        assert new_size > int(
+            pvc_size
+        ), f"New size of the PVC cannot be less than existing size: new size is {new_size})"
+
+        baseui_obj = PageNavigator(setup_ui)
```
Comment (OdedViner) [inline tests/ui/test_pvc_ui.py:180]: @am-agrawa why do you use `PageNavigator` class here?

File: tests/ui/test_pvc_ui.py [hunk 11]
```diff
@@ -130,11 +145,63 @@ def test_create_delete_pvc(
...
+        expected_capacity = baseui_obj.wait_until_expected_text_is_found(
```
Comment (OdedViner) [inline tests/ui/test_pvc_ui.py:181]: you can add method on `pvc_ui` class

Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: done

File: tests/ui/test_pvc_ui.py (hunk 10 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: removed

File: ocs_ci/ocs/ui/base_ui.py (hunk 9 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py]: @ebenahar minor code change, please review

File: ocs_ci/ocs/ui/base_ui.py [hunk 12]
```diff
@@ -36,6 +42,7 @@ class BaseUI:
     """
 
     def __init__(self, driver):
+
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/base_ui.py:45]: why did you add empty line?

File: ocs_ci/ocs/ui/base_ui.py [hunk 13]
```diff
@@ -204,6 +258,10 @@ class PageNavigator(BaseUI):
 
     def __init__(self, driver):
         super().__init__(driver)
+        ocp_version = get_ocp_version()
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/base_ui.py:261]: Why did you add these lines?

File: ocs_ci/ocs/ui/base_ui.py [hunk 14]
```diff
@@ -195,6 +202,53 @@ def take_screenshot(self):
...
+            poll_frequency=1,
+            ignored_exceptions=[
+                NoSuchElementException,
+                ElementNotVisibleException,
+                StaleElementReferenceException,
+                TimeoutException,
+            ],
+        )
+        wait.until(ec.visibility_of_element_located((locator[1], locator[0])))
+        text_found = (self.driver.find_element(locator[1], locator[0])).text
```
Comment (shyRozen) [inline ocs_ci/ocs/ui/base_ui.py:247]: Can be done with one command:
wait.until(ec.text_to_be_present_in_element((locator[1], locator[0]), expected_text))

File: ocs_ci/ocs/ui/base_ui.py [hunk 15]
```diff
@@ -195,6 +202,53 @@ def take_screenshot(self):
...
+        assert (
+            expected_text == text_found
+        ), f"Text found: {text_found}, Expected text: {expected_text}"
+
```
Comment (shyRozen) [inline ocs_ci/ocs/ui/base_ui.py:251]: A match shorter implementation which I did but didn't sent the PR yet.

```
wait = WebDriverWait(self.driver, timeout)
        try:
            wait.until(ec.text_to_be_present_in_element((locator[1], locator[0]), expected_text))
            return True
        except TimeoutException as e:
            self.take_screenshot()
            logger.error(f"Locator {locator[1]} {locator[0]} did not find text {expected_text}")
            return False
```
We to tend not to assert in the function but in the test script itself so you can construct more appropriate error

File: ocs_ci/ocs/ui/base_ui.py [hunk 16]
```diff
@@ -195,6 +202,53 @@ def take_screenshot(self):
...
+        Args:
+            locator (tuple): (GUI element needs to operate on (str), type (By))
+            timeout (int): Looks for a web element until timeout (sec) occurs
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((locator[1], locator[0])))
+        element.clear()
+
+    def wait_until_expected_text_is_found(self, locator, expected_text, timeout=300):
```
Comment (shyRozen) [inline ocs_ci/ocs/ui/base_ui.py:218]: Please change the timeout to 10 which will be used for most scenarios and customize it in your call

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 17]
```diff
@@ -65,15 +76,15 @@ def create_pvc_ui(self, sc_type, pvc_name, access_mode, pvc_size, vol_mode):
...
         Verifying PVC details via UI
 
-        pvc_size (str): the size of pvc (GB)
-        access_mode (str): access mode
-        vol_mode (str): volume mode type
+        Args:
+            pvc_size (str): the size of pvc (GB)
+            access_mode (str): access mode
+            vol_mode (str): volume mode type
+            sc_type (str): storage class type
```
Comment (shyRozen) [inline ocs_ci/ocs/ui/pvc_ui.py:87]: The comment is not related to this line but could you please add navigation to the specific pvc page so the function is usable from any spot you call it and not just after pvc creation.

File: ocs_ci/ocs/ui/base_ui.py (hunk 15 above)
Comment (shyRozen) [reply ocs_ci/ocs/ui/base_ui.py:251]: There is issue with wait for element that the element can have different text before it changes to the expected text and this scenario will not catch. Waiting for the text is better.

File: ocs_ci/ocs/ui/base_ui.py (hunk 12 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:45]: removed

File: ocs_ci/ocs/ui/base_ui.py (hunk 13 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:261]: added by mistake, removed

File: ocs_ci/ocs/ui/base_ui.py (hunk 15 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:251]: As suggested, moved the Assertion to test file and the updated code is as below:

```suggestion
      wait = WebDriverWait(
            self.driver,
            timeout=timeout,
            poll_frequency=1,
            ignored_exceptions=[
                NoSuchElementException,
                ElementNotVisibleException,
                StaleElementReferenceException,
                TimeoutException,
            ],
        )
        if wait.until(ec.text_to_be_present_in_element((locator[1], locator[0]), expected_text)):
            text_found = (self.driver.find_element(locator[1], locator[0])).text
            return text_found  wait = WebDriverWait(
            self.driver,
            timeout=timeout,
            poll_frequency=1,
            ignored_exceptions=[
                NoSuchElementException,
                ElementNotVisibleException,
                StaleElementReferenceException,
                TimeoutException,
            ],
        )
        wait.until(ec.visibility_of_element_located((locator[1], locator[0])))
        text_found = (self.driver.find_element(locator[1], locator[0])).text
        assert (
            expected_text == text_found
        ), f"Text found: {text_found}, Expected text: {expected_text}"

```

Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:251]: As suggested, moved the Assertion to test file and the updated code is as below:

```suggestion
      wait = WebDriverWait(
            self.driver,
            timeout=timeout,
            poll_frequency=1,
            ignored_exceptions=[
                NoSuchElementException,
                ElementNotVisibleException,
                StaleElementReferenceException,
                TimeoutException,
            ],
        )
        if wait.until(ec.text_to_be_present_in_element((locator[1], locator[0]), expected_text)):
            text_found = (self.driver.find_element(locator[1], locator[0])).text
            return text_found 
```

File: ocs_ci/ocs/ui/base_ui.py (hunk 16 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:218]: I have kept it to 60 by default

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 17 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:87]: added, please check

File: ocs_ci/ocs/ui/views.py [hunk 18]
```diff
@@ -252,6 +252,23 @@
...
+    "resize-value": ("//input[@name='requestSizeValue']", By.XPATH),
+    "expand-btn": ("#confirm-action", By.CSS_SELECTOR),
+    "pvc-status": (
+        "dd[data-test-id='pvc-status'] span[data-test='status-text']",
+        By.CSS_SELECTOR,
+    ),
+    "test-project-link": ("//a[normalize-space()='{}']", By.XPATH),
+    "expected-capacity": (
+        "//dd[contains(text(),'{}') and @data-test='pvc-requested-capacity']",
+        By.XPATH,
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/views.py:266]: You can use this locator.
``` "expected-capacity": (('//*[@data-test="pvc-requested-capacity"]', By.XPATH) ```

File: ocs_ci/ocs/ui/views.py [hunk 19]
```diff
@@ -252,6 +252,23 @@
...
+    ),
+    "new-capacity": (
+        "//dd[contains(text(),'{}') and @data-test-id='pvc-capacity']",
+        By.XPATH,
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/views.py:270]: You can use this locator.
"expected-capacity": (('//*[@data-test="pvc-capacity"]', By.XPATH)

File: ocs_ci/ocs/ui/views.py (hunk 18 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/views.py:266]: This locator will not work in my test scenario and will return True even if the 'Capacity' value doesn't change after PVC resize.

File: ocs_ci/ocs/ui/views.py (hunk 19 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/views.py:270]: This locator will not work in my test scenario and will return True even if the 'Capacity' value doesn't change after PVC resize.

Comment (OdedViner) [reply ocs_ci/ocs/ui/views.py:270]: ok

File: ocs_ci/ocs/ui/views.py (hunk 18 above)
Comment (OdedViner) [reply ocs_ci/ocs/ui/views.py:266]: ok

File: ocs_ci/ocs/ui/base_ui.py [hunk 20]
```diff
@@ -195,6 +201,52 @@ def take_screenshot(self):
...
+                ElementNotVisibleException,
+                StaleElementReferenceException,
+                TimeoutException,
+            ],
+        )
+        if wait.until(
+            ec.text_to_be_present_in_element((locator[1], locator[0]), expected_text)
+        ):
+            text_found = (self.driver.find_element(locator[1], locator[0])).text
+            return text_found
```
Comment (shyRozen) [inline ocs_ci/ocs/ui/base_ui.py:248]: I think you can finish here the check and compare the expected_text to the text_found and just return true or false. This way every call to the function don't have to deal with another comparison of the text.

File: tests/ui/test_pvc_ui.py [hunk 21]
```diff
@@ -121,20 +137,84 @@ def test_create_delete_pvc(
...
+        self.pvc_loc = locators[ocp_version]["pvc"]
+
+        # Verifying PVC expansion
+        logger.info("Verifying PVC resize")
+        expected_capacity = f"{new_size} GiB"
+        text_found = pvc_ui_obj.wait_until_expected_text_is_found(
+            format_locator(self.pvc_loc["new-capacity"], expected_capacity),
+            expected_text=expected_capacity,
+            timeout=300,
+        )
```
Comment (shyRozen) [inline tests/ui/test_pvc_ui.py:195]: Doesn't `pvc_ui_obj.verify_pvc_resize_ui(expected_capacity=expected_capacity)` after this code is doing the same?

File: tests/ui/test_pvc_ui.py [hunk 22]
```diff
@@ -121,20 +137,84 @@ def test_create_delete_pvc(
...
+
+        pvc_ui_obj.verify_pvc_resize_ui(expected_capacity=expected_capacity)
```
Comment (shyRozen) [inline tests/ui/test_pvc_ui.py:197]: Please add a return to the function verify_pvc_resize_ui True or False if it is verified or not and set here a variable to catch the return result and deal with it.

File: tests/ui/test_pvc_ui.py [hunk 23]
```diff
@@ -121,20 +137,84 @@ def test_create_delete_pvc(
...
+
+        assert (
+            expected_capacity == text_found
```
Comment (shyRozen) [inline tests/ui/test_pvc_ui.py:200]: Please compare this in the function itself and return true or false.

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 17 above)
Comment (shyRozen) [reply ocs_ci/ocs/ui/pvc_ui.py:89]: Great

File: ocs_ci/ocs/ui/base_ui.py (hunk 20 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py]: please check

File: tests/ui/test_pvc_ui.py (hunk 21 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: removed

File: tests/ui/test_pvc_ui.py (hunk 22 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: please check

File: tests/ui/test_pvc_ui.py (hunk 23 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: please check

File: ocs_ci/ocs/ui/base_ui.py [hunk 24]
```diff
@@ -195,6 +201,59 @@ def take_screenshot(self):
...
+        """
+        wait = WebDriverWait(
+            self.driver,
+            timeout=timeout,
+            poll_frequency=1,
+            ignored_exceptions=[
+                NoSuchElementException,
+                ElementNotVisibleException,
+                StaleElementReferenceException,
+                TimeoutException,
```
Comment (shyRozen) [inline ocs_ci/ocs/ui/base_ui.py:240]: You have to take this one out if using it below

File: tests/ui/test_pvc_ui.py [hunk 25]
```diff
@@ -92,18 +100,34 @@ def teardown(self):
             ),
         ],
     )
-    def test_create_delete_pvc(
-        self, setup_ui, sc_type, pvc_name, access_mode, pvc_size, vol_mode
+    def test_create_resize_delete_pvc(
+        self,
+        project_factory,
+        pod_factory,
```
Comment (shyRozen) [inline tests/ui/test_pvc_ui.py:106]: Need to be removed.

File: ocs_ci/ocs/ui/base_ui.py (hunk 24 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py]: removed

File: tests/ui/test_pvc_ui.py (hunk 25 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: Removed

**Comment by am-agrawa** [comment]
@OdedViner @ebondare @ebenahar PR validation is successful, please approve and merge the PR.

File: ocs_ci/ocs/ui/ui_utils.py [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+    Args:
+        locator (tuple): (GUI element needs to operate on (str), type (By))
+        string_to_insert (str): Name of the variable (string) which contains the dynamic web element
+            when generated on certain action
+
+    return:
+        formats the locator using .format() function which takes string to be inserted as an argument
+
+    """
+    return locator[0].format(string_to_insert), locator[1]
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/ui_utils.py:14]: Can move this function to `ocs_ci/ocs/ui/helpers_ui.py`?

File: ocs_ci/helpers/helpers.py [hunk 27]
```diff
@@ -203,7 +203,7 @@ def create_pod(
             ] = pvc_name
 
     if interface_type == constants.CEPHBLOCKPOOL and raw_block_pv:
-        if pod_dict_path in [constants.FEDORA_DC_YAML, constants.FIO_DC_YAML]:
+        if pod_dict in [constants.FEDORA_DC_YAML, constants.FIO_DC_YAML]:
             temp_dict = [
```
Comment (OdedViner) [inline ocs_ci/helpers/helpers.py:207]: Why did you change the dictionary name `pod_dict_path`?

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 28]
```diff
@@ -87,28 +121,103 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
...
+        """
+        Resizing pvc via UI
+
+        Args:
+            project_name (str): name of test project
+            pvc_name (str): the name of pvc
+            new_size (int): the new size of pvc (GB)
+
+        """
+
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/pvc_ui.py:143]: Can remove this empty line?

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 29]
```diff
@@ -87,28 +121,103 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
...
+    def verify_pvc_resize_ui(self, expected_capacity):
+        """
+        Verifying PVC resize via UI
+
+        Args:
+            expected_capacity (str): the new size of pvc (GiB)
+
+        """
+
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/pvc_ui.py:180]: Please add "navigate" method at the beginning of this function...

File: tests/ui/test_pvc_ui.py [hunk 30]
```diff
@@ -126,20 +149,91 @@ def test_create_delete_pvc(
...
+
+        assert pvc_resize, "PVC resize failed"
+        logger.info(
+            "Pvc resize verified..!!"
+            f"New Capacity after PVC resize is {expected_capacity}"
+        )
+
+        # Running FIO
+        logger.info("Execute FIO on a Pod")
+        if vol_mode in constants.VOLUME_MODE_BLOCK:
```
Comment (OdedViner) [inline tests/ui/test_pvc_ui.py:216]: constants.VOLUME_MODE_BLOCK is string(no list)
you can change to `vol_mode==constants.VOLUME_MODE_BLOCK`

File: ocs_ci/helpers/helpers.py (hunk 27 above)
Comment (am-agrawa) [reply ocs_ci/helpers/helpers.py:207]: This change was suggested by Jilju for the issue- 2524.

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 28 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:143]: ok

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 29 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:180]: added

File: ocs_ci/ocs/ui/base_ui.py [hunk 31]
```diff
@@ -196,6 +199,53 @@ def take_screenshot(self):
...
+    def wait_until_expected_text_is_found(self, locator, expected_text, timeout=60):
+        """
+        Method to wait for a expected text to appear on the UI (use of explicit wait type),
+        this method is helpful in working with elements which appear on completion of certain action and
+        ignores all the listed exceptions for the given timeout.
+
+        Args:
+            locator (tuple): (GUI element needs to operate on (str), type (By))
+            expected_text (str): Text which needs to be searched on UI
+            timeout (int): Looks for a web element repeatedly until timeout (sec) occurs
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/base_ui.py:224]: empty line is missing below

File: ocs_ci/ocs/ui/ui_utils.py (hunk 26 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/ui_utils.py:14]: moved and deleted ui_utils.py file

File: tests/ui/test_pvc_ui.py (hunk 30 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py:216]: thanks for pointing out, updated

File: ocs_ci/ocs/ui/base_ui.py (hunk 31 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/base_ui.py:224]: added

File: tests/ui/test_pvc_ui.py [hunk 32]
```diff
@@ -20,14 +24,18 @@ class TestPvcUserInterface(object):
...
     """
 
-    def teardown(self):
-        pvc_objs = get_all_pvc_objs(namespace="openshift-storage")
-        pvcs = [pvc_obj for pvc_obj in pvc_objs if "test-pvc" in pvc_obj.name]
-        delete_pvcs(pvc_objs=pvcs)
+    @pytest.fixture()
+    def teardown(self, request):
+        def finalizer():
+            pvc_objs = get_all_pvc_objs(namespace="openshift-storage")
```
Comment (prsurve) [inline tests/ui/test_pvc_ui.py:30]: make use of constant values

File: tests/ui/test_pvc_ui.py [hunk 33]
```diff
@@ -126,20 +149,91 @@ def test_create_delete_pvc(
...
+            "Pvc resize verified..!!"
+            f"New Capacity after PVC resize is {expected_capacity}"
+        )
+
+        # Running FIO
+        logger.info("Execute FIO on a Pod")
+        if vol_mode in constants.VOLUME_MODE_BLOCK:
+            storage_type = "block"
+        else:
+            storage_type = "fs"
```
Comment (prsurve) [inline tests/ui/test_pvc_ui.py:219]: make use of constants WORKLOAD_STORAGE_TYPE_BLOCK,WORKLOAD_STORAGE_TYPE_FS

File: ocs_ci/helpers/helpers.py (hunk 27 above)
Comment (prsurve) [reply ocs_ci/helpers/helpers.py:207]: if this is to resolve issue make sure to add Fixes in PR

File: tests/ui/test_pvc_ui.py (hunk 33 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py:219]: changed

File: ocs_ci/helpers/helpers.py (hunk 27 above)
Comment (am-agrawa) [reply ocs_ci/helpers/helpers.py]: thanks @prsurve

File: tests/ui/test_pvc_ui.py (hunk 32 above)
Comment (am-agrawa) [reply tests/ui/test_pvc_ui.py]: teardown removed

**Review by OdedViner — APPROVED** [review]
LGTM

**Review by ebenahar — APPROVED** [review]
